src/storage/src/source/reclock/compat.rs

```diff
@@ -109,6 +115,9 @@ pub struct PersistHandle {
     native_source_upper: MutableAntichain<Partitioned<PartitionId, MzOffset>>,
     compat_source_upper: MutableOffsetAntichain,
     minimum_produced: bool,
+    // We hold on to this in case we need to create a `ReadHandle` for fetching
+    // an up-to-date snapshot.
+    persist_client: PersistClient,
```

Comment:
@danhhz Is it cool to hold on to a `PersistClient` like this, instead of an `Arc<Mutex<PersistClientCache>>`?

Reply:
yup! we only pass around `Arc<Mutex<PersistClientCache>>` in dataflow operators because we don't have a nice place to run the async code during operator initialization